clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12408,6 +12408,29 @@ def err_sycl_special_type_num_init_method : Error<
   "types with 'sycl_special_class' attribute must have one and only one '__init' "
   "method defined">;
 
+// SYCL kernel entry point diagnostics
+def err_sycl_entry_point_invalid : Error<
+  "'sycl_kernel_entry_point' attribute cannot be applied to a"
+  " %select{non-static member|variadic|deleted|defaulted|constexpr|consteval|"
+           "noreturn|coroutine}0 function">;
```

Comment:
is `coroutine function` the term of art?  I thought it was just `coroutine`?
```suggestion
           "noreturn|coroutine}0%select{||||||| function}0">;
```

Reply:
Yeah, that extra select is kind of ugly, I agree.  The alternative is to repeat ` function` on all the other alternatives.

Reply:
Done. I avoided the additional `%select` by duplicating " function" for the relevant cases. I also changed the wording for the noreturn case from "no return function" to "function declared with the 'noreturn' attribute".